### PR TITLE
Removing all static registration in favor of instance-based

### LINKED ISSUE #25: Please add support for calling AddAutoMapper() multiple times
Please consider adding support for calling `AddAutoMapper()` multiple times.

Reason:
We have very modular application design, each module is responsible of registering own mapper profiles.
Currently `services.AddAutoMapper(i => i.AddProfile(...));` cannot be called multiple times because of `Mapper.Initialize` being overwriting previous setup. Additionally services are not added with `TryAdd...`

### LINKED ISSUE #28: .net core 2.0 xunit integration test
Hi, i have an error after upgrade to .net core 2.0 from 1.1 when try run integration test with xunit framework, but when i comment `services.AddAutoMapper();` in the `ConfigureServices(IServiceCollection services)` from `Startup` class it works fine
```
 public class ThingsController_Tests
    {
        private readonly HttpClient _client;
        public ThingsController_Tests()
        {
            var hostBuilder = new WebHostBuilder();

            
            var server = new TestServer(hostBuilder.UseStartup<Startup>());
            this._client = server.CreateClient();
        }
...
```

packages from my test project, it has link to main project
```
<PackageReference Include="Microsoft.AspNetCore.TestHost" Version="2.0.0" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
    <PackageReference Include="xunit.runner.console" Version="2.2.0" />
    <PackageReference Include="xunit.runner.visualstudio" Version="2.2.0" />
    <PackageReference Include="xunit" Version="2.2.0" />
    <PackageReference Include="xunit.runners" Version="2.0.0" />
```

automapper packages in main project 
 ```
 <PackageReference Include="automapper" Version="6.1.1" />
    <PackageReference Include="AutoMapper.Extensions.Microsoft.DependencyInjection" Version="3.0.1" />
```
help pls :)
```
at System.Reflection.RuntimeModule.GetTypes(RuntimeModule module)
   at System.Reflection.RuntimeAssembly.get_DefinedTypes()
   at System.Linq.Enumerable.SelectManySingleSelectorIterator`2.ToArray()
   at System.Linq.Enumerable.ToArray[TSource](IEnumerable`1 source)
   at AutoMapper.ServiceCollectionExtensions.AddAutoMapperClasses(IServiceCollection services, Action`1 additionalInitAction, IEnumerable`1 assembliesToScan)
   at myProjectName.Startup.ConfigureServices(IServiceCollection services) in myProjectPath
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at Microsoft.AspNetCore.Hosting.ConventionBasedStartup.ConfigureServices(IServiceCollection services)
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.EnsureApplicationServices()
   at Microsoft.AspNetCore.Hosting.Internal.WebHost.BuildApplication()
   at Microsoft.AspNetCore.Hosting.WebHostBuilder.Build()
   at Microsoft.AspNetCore.TestHost.TestServer..ctor(IWebHostBuilder builder, IFeatureCollection featureCollection)
   at Tests_myprojectName.ThingsController_Tests..ctor() in myProjectPath
Result Message:	System.Reflection.ReflectionTypeLoadException : Unable to load one or more of the requested types.
```

### LINKED ISSUE #36: Extension Doesn't Work with AutoMapper 6.2.0 In Integration Tests
When [integration testing with ASP.Net Core](https://docs.microsoft.com/en-us/aspnet/core/testing/integration-testing), it is very common to use `TestServer` to start your service as an in-memory host so that you can perform service-level integration tests via a real `HttpClient`.

However, since AutoMapper 6.2.0 was released this no longer works. Your `Startup` class is effectively called multiple times within the same process, so the `AddAutoMapper()` call happens multiple times, so the `Mapper.Initialize()` call also happens multiple times. This results in an exception in the second test:

```
System.InvalidOperationException : Mapper already initialized. You must call Initialize once per application domain/process.
   at AutoMapper.Mapper.set_Configuration(IConfigurationProvider value)
   at AutoMapper.Mapper.Initialize(Action`1 config)
   at AutoMapper.ServiceCollectionExtensions.AddAutoMapperClasses(IServiceCollection services, Action`1 additionalInitAction, IEnumerable`1 assemblies)
   at MyService.Startup.ConfigureServices(IServiceCollection services)
```

How can we 'clear out' the AutoMapper statics so that integration testing can be supported in a single process?

### LINKED ISSUE #49: services.AddAutoMapper() breaks EF Core CLI
When I run the `dotnet ef database drop` command, it throws exception:

```
An error occurred while calling method 'BuildWebHost' on class 'Program'. Continuing without the application service provider. Error: Mapper already initialized. You must call Initialize once per application domain/process.
Unable to create an object of type 'ApplicationDbContext'. Add an implementation of 'IDesignTimeDbContextFactory<ApplicationDbContext>' to the project, or see https://go.microsoft.com/fwlink/?linkid=851728 for additional patterns supported at design time.
```

It seems to caused by `services.AddAutoMapper()` in ASP.NET Core `ConfigureServices` method. When I comment this line, command works fine. I tried other commends like database update or migrations, this problem won't happen.

Solution from #36 works. But I think the default extension method shouldn't break EF Core CLI.

I'm using AutoMapper 6.2.2 and AutoMapper.Extensions.Microsoft.DependencyInjection 3.2.0, with Entity Framework Core 2.1.0-preview1-final.

### LINKED ISSUE #55: AddAutoMapper and WebApplicationFactory do not play nice.
[Sample app.](https://github.com/VictorioBerra/AutoMapper.XUnit.MapperAlreadyInitialized)

I tried to use the absolute bare minimum code from basically `dotnet new mvc && dotnet new xunit` where I `AddAutoMapper` and I also have two tests class using a pretty much empty `WebApplicationFactory`. I also tossed a single profile in the mvc app. This was incredibly easy to reproduce.

`Mapper.Reset();` in my startup did not help, as well as disabling `UseStaticRegistration`. 

If I just use a single `WebApplicationFactory` then my tests do not fail so is this some weird thread issue with `IClassFixture`? From my research there should not be two separate instances of a fixture created in xunit but this could be wrong. Somehow, my startup is being initialized multiple times.

Any help is appreciated. I feel like this is an important issue because the official Microsoft tutorial and example shows how to write tests using the new WebApplicationFactory as shown [here](https://docs.microsoft.com/en-us/aspnet/core/test/integration-tests?view=aspnetcore-2.1) and anyone with two XUnit test classes inheriting from their implementation of the factory will run into this very error and their tests will fail. For what its worth, I am not running my tests in parallel.

ref https://github.com/AutoMapper/AutoMapper.Extensions.Microsoft.DependencyInjection/pull/26#issuecomment-404074972

### LINKED ISSUE #56: Remove option to register statically and do instance-only
Big problems with registering statically:

`services.AddAutoMapper` is called on an instance of a `IServiceCollection`, but `Mapper.Initialize` is static. I can't know if `AddAutoMapper` has already been called with against the same `IServiceCollection` instance. Even if I track if I've already initialized AutoMapper statically, I can't know if it's against the same set of services.

Just far more trouble than it's worth, I think.

### PR DESCRIPTION
**Pull request by jbogard**
Fixes #36; Fixes #56; Fixes #55; Fixes #49; Fixes #25; Fixes #28

### CODE REVIEW

File: src/AutoMapper.Extensions.Microsoft.DependencyInjection/AutoMapper.Extensions.Microsoft.DependencyInjection.csproj [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
-        <Authors>Jimmy Bogard</Authors>
-        <TargetFrameworks>netstandard2.0</TargetFrameworks>
-        <AssemblyName>AutoMapper.Extensions.Microsoft.DependencyInjection</AssemblyName>
-        <PackageId>AutoMapper.Extensions.Microsoft.DependencyInjection</PackageId>
-        <PackageIconUrl>https://s3.amazonaws.com/automapper/icon.png</PackageIconUrl>
-        <PackageProjectUrl>http://automapper.org</PackageProjectUrl>
-        <PackageLicenseUrl>https://github.com/AutoMapper/AutoMapper/blob/master/LICENSE.txt</PackageLicenseUrl>
-        <GenerateAssemblyVersionAttribute>false</GenerateAssemblyVersionAttribute>
-    </PropertyGroup>
+  <PropertyGroup>
```
Comment (lbargaoanu) [inline src/AutoMapper.Extensions.Microsoft.DependencyInjection/AutoMapper.Extensions.Microsoft.DependencyInjection.csproj]: Add SourceLink?
